src/main/scala/org/apache/spark/deploy/armada/submit/PodMerger.scala

```diff
@@ -0,0 +1,90 @@
+  *
+  * Key insight: PodBuilder(pod) preserves ALL fields automatically. We only override the 3 arrays
+  * that need strategic merge by name. Everything else (40+ fields) is handled generically by
+  * fabric8.
+  *
+  * This solves the duplicate volume/initContainer bug via mergeByName.
+  */
+private[submit] object PodMerger {
+
+  /** Strategic merge of two fabric8 Pods.
+    *
+    * Arrays (volumes, containers, initContainers) merge by name - overriding wins for duplicates.
+    * All other fields automatically preserved from overriding Pod via PodBuilder.
+    *
+    * This is truly generic - works for ANY Pod fields without enumeration!
+    */
+  def mergePods(base: Pod, overriding: Pod): Pod = {
+    val baseSpec = Option(base.getSpec).getOrElse(new io.fabric8.kubernetes.api.model.PodSpec())
+    val overridingSpec =
+      Option(overriding.getSpec).getOrElse(new io.fabric8.kubernetes.api.model.PodSpec())
+
+    // Strategic merge for named arrays - prevents duplicates!
+    val mergedVolumes = mergeByName(
+      Option(baseSpec.getVolumes).map(_.asScala.toSeq).getOrElse(Seq.empty),
+      Option(overridingSpec.getVolumes).map(_.asScala.toSeq).getOrElse(Seq.empty)
+    )(v => Option(v.getName))
+
+    val mergedInitContainers = mergeByName(
+      Option(baseSpec.getInitContainers).map(_.asScala.toSeq).getOrElse(Seq.empty),
+      Option(overridingSpec.getInitContainers).map(_.asScala.toSeq).getOrElse(Seq.empty)
+    )(c => Option(c.getName))
+
+    val mergedContainers = mergeByName(
+      Option(baseSpec.getContainers).map(_.asScala.toSeq).getOrElse(Seq.empty),
+      Option(overridingSpec.getContainers).map(_.asScala.toSeq).getOrElse(Seq.empty)
+    )(c => Option(c.getName))
+
+    // Merge nodeSelectors (Map merge)
+    val baseNodeSelector =
+      Option(baseSpec.getNodeSelector).map(_.asScala.toMap).getOrElse(Map.empty)
+    val overridingNodeSelector =
+      Option(overridingSpec.getNodeSelector).map(_.asScala.toMap).getOrElse(Map.empty)
+    val mergedNodeSelector = baseNodeSelector ++ overridingNodeSelector
+
+    // PodBuilder magic: preserves ALL fields from overriding, we only override what we merged
+    new PodBuilder(overriding)
+      .editOrNewSpec()
+      .withVolumes(mergedVolumes.asJava)
+      .withInitContainers(mergedInitContainers.asJava)
+      .withContainers(mergedContainers.asJava)
+      .withNodeSelector(if (mergedNodeSelector.nonEmpty) mergedNodeSelector.asJava else null)
+      .endSpec()
+      .build()
+  }
+
+  /** Merge arrays by name - simple strategic merge. Items from 'overriding' replace items from
+    * 'base' with same name.
+    */
+  def mergeByName[T](base: Seq[T], overriding: Seq[T])(
+      getName: T => Option[String]
+  ): Seq[T] = {
+    val overridingNames = overriding.flatMap(getName(_).toSeq).toSet
+    val uniqueBase      = base.filterNot(item => getName(item).exists(overridingNames.contains))
+    uniqueBase ++ overriding
+  }
```

Comment:
This:
```suggestion
  def mergePartByName[T](getPart: Pod => T)(basePod: Pod, overridingPod: Pod)(
      getName: T => Option[String]
  ): Seq[T] = {
    val base = getPart(basePod).map(_.asScala.toSeq).getOrElse(Seq.empty)
    val overriding = getPart(overridingPod).map(_.asScala.toSeq).getOrElse(Seq.empty)
    val overridingNames = overriding.flatMap(getName(_).toSeq).toSet
    val uniqueBase      = base.filterNot(item => getName(item).exists(overridingNames.contains))
    uniqueBase ++ overriding
  }
```

would allow for this simplification above
```diff
-    val mergedVolumes = mergeByName(
-      Option(baseSpec.getVolumes).map(_.asScala.toSeq).getOrElse(Seq.empty),
-      Option(overridingSpec.getVolumes).map(_.asScala.toSeq).getOrElse(Seq.empty)
+    val mergedVolumes = mergePartByName(_.getVolumes)(
+      baseSpec, overridingSpec
     )(v => Option(v.getName))
```